src/dmd/expressionsem.d

```diff
@@ -5129,12 +5129,40 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             }
             else if (sc.func)
             {
+                FuncDeclaration fd      = s.isFuncDeclaration();
+                AggregateDeclaration ag = s.isAggregateDeclaration();
+                EnumDeclaration ed      = s.isEnumDeclaration();
+                bool inserted     =  (fd || ag || ed || (v && v.isDataseg())) && sc.func.localsymtab.insert(s);
+                //  Try to insert static decl from subscope, see https://issues.dlang.org/show_bug.cgi?id=issue 3031
+                //  ---------------------------------------------------
+                //  let the func body `{ { static T decl; } { static T decl; } }`
+                //  second `decl` insertion fails.
+                //  the fix is to replace the static decl name and use the old sym name as **local** alias,
+                //  to get            `{ { static T decl; } { static T RndID; alias decl = RndID; }`
+                //  as 1) it is not possible to link against a global var that is hidden in a func body.
+                //     2) in a func body, out of order decls are not supported so we can patch the name without affecting prior code
+                if (!inserted && (sc.flags & SCOPE.subfuncscope) &&
+                    ((v && v.isDataseg()) || ed || (fd && fd.storage_class & STC.static_)))
+                {
+                    // printf("fixing 3031, on %s at line %d\n", s.toChars(), s.loc.linnum);
+                    Identifier oldID    = s.ident;
+                    s.ident             = Identifier.generateId("issue3031_");
```

Comment:
This name is unrelated to the original, how on earth are you supposed to debug print nested static symbols?

Reply:
true in theory but in practice debugging local statics does not work
![thory_vs_hjkl](https://user-images.githubusercontent.com/65831385/84153700-7f754780-aa66-11ea-8f99-f1b35e7bb640.png)

Reply:
Maybe rename those variables as `<original name>_<line>_<column>`?

Reply:
yes good idea.

Reply:
On Windows, debug info for statics exists and can be used in watches, but they are not enumerated with the locals (I guess it is the same for ELF object files). LDC/GDC might be different, too.

The `<original name>_<line>_<column>` pattern isn't good enough for templates, I think your test case should expose this with a third instantiation.

An alternative might be to keep the identifier as is, but to insert a parent symbol for the scope in the mangled name.

Reply:
> true in theory but in practice debugging local statics does not work

It has worked in gdb in the past.